### PR TITLE
fix: mergefix MM impersonator tightened checks

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

tl;dr https://github.com/shapeshift/web/pull/5292 brought some naive, immediate checks unborking native Cosmos staking for snaps, which were superseeded by https://github.com/shapeshift/web/pull/5287 tightening the checks, in particular for MM impersonators which appear as `isMetaMask`.
Release mergy shenanigans ensued, and we lost the https://github.com/shapeshift/web/commit/04557416d20c20165482334a3620f540df299f5f diff by bringing https://github.com/shapeshift/web/pull/5292 *after* https://github.com/shapeshift/web/pull/5287. https://github.com/shapeshift/web/pull/5314 did fix things in some fashion, though without the tightened checks for Brave.

This diffes the diff to make the diff diffed again.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

N/A

## Risk

None, this was already stamped and tested

## Testing

- Cosmos native staking is still working with MM snaps
- Brave is still happy, without console spew

### Engineering

- ☝🏽 
- Ensure WC sends/savers native deposits/withdraws are still happy against *develop* as per https://github.com/shapeshift/web/pull/5314

### Operations

- ☝🏽 

## Screenshots (if applicable)

- Snaps cosmos staking

image
image

- Brave with no spew

image

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #5321** <a href="https://app.graphite.dev/github/pr/shapeshift/web/5321" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/5321?utm_source=stack-comment).

**Review by 0xApotheosis — APPROVED** [review]
The diff is great again 🥳
